src/main/java/net/tirasa/connid/bundles/ad/authentication/ADAuthenticate.java

```diff
@@ -140,4 +139,18 @@ public Uid resolveUsername() {
         }
         return authnObject.getUid();
     }
+
+    private static boolean isNotBlankCredentials(final GuardedString credentials) {
```

Comment:
With such approach you are basically duplicating the same `GuardedString` access performed elsewhere in the same class: can't we do that once, instead?

Reply:
Yes, the other access was too "inner" so I opted for this solution. Here https://github.com/Tirasa/ConnIdADBundle/blob/master/src/main/java/net/tirasa/connid/bundles/ad/ADConnection.java#L204-L213

But I think that is doable. let me work on it.